external-crates/move/move-compiler/src/shared/mod.rs

```diff
@@ -219,16 +347,13 @@ impl CompilationEnv {
         if !is_filtered {
             // add help to suppress warning, if applicable
             // TODO do we want a centralized place for tips like this?
-            if diag.info().severity() == Severity::Warning && !diag.info().is_external() {
-                let possible_filter = WarningFilter::Code(
-                    Category::try_from(diag.info().category()).unwrap(),
-                    diag.info().code(),
-                );
-                if let Some(filter_name) = possible_filter.to_str() {
+            if diag.info().severity() == Severity::Warning {
+                if let Some(filter_name) = self.known_filter_names.get(&diag.info().id()) {
                     let help = format!(
                         "This warning can be suppressed with '#[{}({})]' \
                         applied to the 'module' or module member ('const', 'fun', or 'struct')",
-                        WARNING_FILTER_ATTR, filter_name
+                        WARNING_FILTER_ATTR,
```

Comment:
Just noting again that we need the prefix here now

Reply:
Noted!